src/libsync/logger.cpp

```diff
@@ -80,6 +80,12 @@ void Logger::doLog(QtMsgType type, const QMessageLogContext &ctx, const QString
             if (_doFileFlush)
                 _logstream->flush();
         }
+#if defined(Q_OS_WIN)
+        if (IsDebuggerPresent()) {
```

Comment:
You may want to extend this check a bit more with other Win32 API debugger checks, e.g., `CheckRemoteDebuggerPresent`.